build.gradle

```diff
@@ -435,6 +435,12 @@ subprojects {
       maxRetries = userMaxTestRetries
       maxFailures = userMaxTestRetryFailures
     }
+
+    // Allows devs to run tests in a loop to debug flaky tests
+    // Eg: I=0; while ./gradlew :streams:test -Prerun-tests --tests org.apache.kafka.streams.integration.IQv2StoreIntegrationTest --fail-fast; do (( I=$I+1 )); echo
+    if (project.hasProperty("rerun-tests")) {
+      outputs.upToDateWhen { false }
+    }
```

Comment:
> which causes us to recompile the tests on every iteration.

just curious. on my local, `cleanTest` does not invoke recompile. The following console output is produced by command `./gradlew cleanTest clients:test --tests RequestResponseTest --info`
```
> Task :clients:compileTestJava UP-TO-DATE
Caching disabled for task ':clients:compileTestJava' because:
  Build cache is disabled
Skipping task ':clients:compileTestJava' as it is up-to-date.
Resolve mutations for :clients:testClasses (Thread[Execution worker Thread 6,5,main]) started.
:clients:testClasses (Thread[Execution worker Thread 6,5,main]) started.

> Task :clients:testClasses UP-TO-DATE
Skipping task ':clients:testClasses' as it has no actions.
Resolve mutations for :clients:checkstyleTest (Thread[Execution worker Thread 6,5,main]) started.
:clients:checkstyleTest (Thread[Execution worker Thread 6,5,main]) started.
```
Could you share the command which causes the recompile to me? thanks.